google/cloud/spanner_dbapi/cursor.py

```diff
@@ -258,9 +259,21 @@ def executemany(self, operation, seq_of_params):
 
         many_result_set = StreamedManyResultSets()
 
-        for params in seq_of_params:
-            self.execute(operation, params)
-            many_result_set.add_iter(self._itr)
+        if classification == parse_utils.STMT_INSERT:
+            match = RE_INSERT.search(operation)
+
+            table_name = match["table_name"].strip("`")
+
+            transaction = self.connection.transaction_checkout()
+            transaction.insert(
+                table=table_name,
+                columns=[col.strip('" ') for col in match["columns"].split(",")],
```

Comment:
This could fail if the SQL string contains comments that include a " or other special characters. Also, if the SQL string contains string literals with quotes or other special characters, this will also fail. So I think this should be considered a 'best-effort' that works with simple SQL strings that are generated by frameworks like SQLAlchemy, but it should not be something that all users **must** use for everything.

Reply:
Yeah, there was a couple of edge cases, I've added some changes today and a test to check that column are designated correctly.